src/models/clustering.jl

```diff
@@ -378,11 +459,83 @@ SpectralClustering
+#     n_best::Int           = 3
+#     svd_method::String    = "randomized"::(_ in ("arpack", "randomized"))
+#     n_svd_vecs::Option{Int} = nothing
+#     mini_batch::Bool      = false
+#     init::Union{AbstractArray,String} = "k-means++"::(_ isa AbstractArray || _ in ("k-means++", "random"))
+#     n_init::Int           = 10::(_ ≥ 1)
+#     random_state::Any     = nothing
 # end
-#
+# function MMI.fitted_params(m::SpectralBiclustering, f)
+#     return (
+#         rows            = pyconvert(Array, f.rows_),
+#         columns         = pyconvert(Array, f.columns_),
+#         row_labels      = pyconvert(Array, f.row_labels_),
+#         column_labels   = pyconvert(Array, f.column_labels_)
+#     )
+# end
+# meta(SpectralBiclustering,
+#     input   = Table(Continuous),
+#     weights = false
+#     )
+
+# """
+# $(MMI.doc_header(SpectralBiclustering))
+
+# Partitions rows and columns under the assumption that the data 
+# has an underlying checkerboard structure. For instance, if there 
+# are two row partitions and three column partitions, each row will 
+# belong to three biclusters, and each column will belong to two 
+# biclusters. The outer product of the corresponding row and column 
+# label vectors gives this checkerboard structure.
+
+# """
+# SpectralBiclustering
+
 # # ============================================================================
-# SpectralCoclustering_ = skcl(:SpectralCoclustering)
+# const SpectralCoclustering_ = skcl(:SpectralCoclustering)
 # @sk_uns mutable struct SpectralCoclustering <: MMI.Unsupervised
+#     n_clusters::Int       = 3::(_ ≥ 1)
+#     svd_method::String    = "randomized"::(_ in ("arpack", "randomized"))
+#     n_svd_vecs::Option{Int} = nothing
+#     mini_batch::Bool      = false
+#     init::Union{AbstractArray,String} = "k-means++"::(_ isa AbstractArray || _ in ("k-means++", "random"))
+#     n_init::Int           = 10::(_ ≥ 1)
+#     random_state::Any     = nothing
 # end
+# function MMI.fitted_params(m::SpectralCoclustering, f)
+#     return (
+#         rows            = pyconvert(Array, f.rows_),
+#         columns         = pyconvert(Array, f.columns_),
+#         row_labels      = pyconvert(Array, f.row_labels_),
+#         column_labels   = pyconvert(Array, f.column_labels_),
+#         biclusters      = Tuple(pyconvert(Array, i) for i in f.biclusters_)
+#     )
+# end
+# meta(SpectralCoclustering,
+#     input   = Table(Continuous),
+#     weights = false
+#     )
+
+# """
+# $(MMI.doc_header(SpectralCoclustering))
+
+# Clusters rows and columns of an array `X` to solve the 
+# relaxed normalized cut of the bipartite graph created 
+# from `X` as follows: the edge between row vertex `i` and 
+# column vertex `j` has weight `X[i, j]`.
+
+# The resulting bicluster structure is block-diagonal, since 
+# each row and each column belongs to exactly one bicluster.
+
+# Supports sparse matrices, as long as they are nonnegative.
+
+# """
+# SpectralCoclustering
+
```

Comment:
I take it the commenting out here is intentional, ie, some future review is needed to see if interface really works here?

Reply:
Yes, that was intentional. `SpectralBiclustering` and `SpectralCoclustering` have `fit` methods, but there is no `predict`/`transform` method for either one. I added them incase we find a good way to integrate them in the future.